zombies.js

```diff
@@ -158,6 +233,20 @@
  * @name equip
  * @param {Weapon} itemToEquip  The weapon item to equip.
  */
+equip(itemsToEquip) {
+  let index = this._pack.indexOf(itemsToEquip);
+  
+  if(itemsToEquip instanceof Weapon && index !== -1) {
+    if(!this.equipped) {
+      this.equipped = itemsToEquip;
+      this._pack.splice(index, 1);
+    } else {
+      this._pack.push(this.equipped);
+      this.equipped = itemsToEquip;
+      this._pack.splice(index, 1);
```

Comment:
You could leverage the `.splice` api to both remove something and replace it with something else.